### PR TITLE
*Fixes problems with Julia 0.4 breaking changes - tupples, dicts,

### PR DESCRIPTION
**Pull request by Petr-Hlavenka**
Uint->UInt, convert(Ptr{UInt32}, foo) -> pointer(foo).
*Changed tests to be runable even without measurement hardware
*tested on NIDAQmx 14.1 + NI USB 6008

### CODE REVIEW

**Comment by bjarthur** [comment]
excellent!  updating to 0.4 has been on my to-do list.  thanks for helping out.

regarding supporting 0.3, i like your suggestion to not support it going forward.  fairly easy with Compat.jl, but it's simply not worth making all the changes to src/functions_Vxxx.jl.  more reasonable is to require a version of NIDAQmx that is a contemporary of the version Julia was on at the time.  so if you want Julia 0.4, then you'll need NIDAQmx 15.1.  if you insist on 14.1, then you're stuck with 0.3.  so i would suggest modifying your PR as follows:
1.  leave the existing src/{functions,constants} unchanged.  
2.  use Clang.jl to add support for NIDAQmx V15.1 instead
3.  add a REQUIRE file which stipulates julia 0.4

i can then tag a new release and publish to METADATA.

i would also urge you to put the changes to the tests (which are great by the way), in a separate PR.  this is unrelated to 0.4 compatibility and so should be kept separate.  since these changes would be useful for 0.3 users, i'd suggest that it should come first.

thanks again petr!

File: src/NIDAQ.jl [hunk 1]
```diff
@@ -41,12 +41,12 @@ export Bool32
 
 try
   global ver
-  major = Uint32[0]
-  ccall((:DAQmxGetSysNIDAQMajorVersion,NIDAQmx),Int32,(Ptr{Uint32},),major)
-  minor = Uint32[0]
-  ccall((:DAQmxGetSysNIDAQMinorVersion,NIDAQmx),Int32,(Ptr{Uint32},),minor)
-  update = Uint32[0]
-  ccall((:DAQmxGetSysNIDAQUpdateVersion,NIDAQmx),Int32,(Ptr{Uint32},),update)
+  major = UInt32[0]
```
Comment (bjarthur) [inline src/NIDAQ.jl:44]: the 0.4 syntax for this is now `foo = Ref{UInt32}(0)`, then `foo[]` to access the returned value.  see http://docs.julialang.org/en/latest/manual/calling-c-and-fortran-code/#passing-pointers-for-modifying-inputs.  also, the corresponding `ccall` needs to specify Ref{UInt32} as the return type.  given that the latter requires extensive changes to src/functions_V*, i'd (again) suggest ditching support for those old versions of NIDAQmx and use Clang.jl to add support for 15.1.

Comment (Petr-Hlavenka) [inline src/NIDAQ.jl:44]: Ben,
could you please help me with installation of Clang.jl. 
I'm getting following error:
'''julia
julia> Pkg.build("Clang")  
INFO: Building Clang  
which: no llvm-config in (/cygdrive/c/Users/phlavenk/AppData/Local/Julia-0.4.0/b
in:/cygdrive/c/Users/phlavenk/AppData/Local/Julia-0.4.0/Git/bin:/cygdrive/c/Prog
ram Files/ConEmu:/cygdrive/c/Program Files/ConEmu/ConEmu:/cygdrive/c/ProgramData
/Oracle/Java/javapath:/cygdrive/c/Program Files/ImageMagick-6.8.9-Q16:/cygdrive/
c/WINDOWS/system32:/cygdrive/c/WINDOWS:/cygdrive/c/WINDOWS/System32/Wbem:/cygdri
ve/c/WINDOWS/System32/WindowsPowerShell/v1.0:/cygdrive/c/Program Files/Intel/WiF
i/bin:/cygdrive/c/Program Files/Common Files/Intel/WirelessCommon:/cygdrive/c/Pr
ogram Files/Microsoft SQL Server/110/Tools/Binn:/cygdrive/c/Program Files (x86)/
CMake/bin:/cygdrive/c/Program Files (x86)/Windows Kits/8.1/Windows Performance T
oolkit:/cygdrive/c/Program Files (x86)/Microsoft SDKs/TypeScript/1.0:/cygdrive/c
/Program Files/Microsoft SQL Server/120/Tools/Binn:/cygdrive/c/Program Files (x8
6)/Git/cmd:/cygdrive/c/Program Files (x86)/Notepad++:/cygdrive/c/Users/phlavenk/
AppData/Local/Continuum/Anaconda:/cygdrive/c/Users/phlavenk/AppData/Local/Contin
uum/Anaconda/Scripts:/cygdrive/c/Program Files (x86)/Tools:/cygdrive/c/Users/phl
avenk/AppData/Local/Julia-0.4.0/bin:/cygdrive/d/Geant4/xerces-c-3.1.1-x86-window
s-vc-10.0/bin:/usr/bin:/cygdrive/c/Program Files/HDF_Group/HDFView/2.11.0/bin:/c
ygdrive/d/Geant4/QT4/bin:/cygdrive/c/root/bin:/cygdrive/c/Users/phlavenk/AppData
/Local/atom/bin)  
================================[ ERROR: Clang ]================================

LoadError: failed process: Process(`which llvm-config`, ProcessExited(1)) [1]  
while loading c:\Users\phlavenk\JuliaData\v0.4\Clang\deps\build.jl, in expressio
n starting on line 12                                                           

# 

================================[ BUILD ERRORS ]================================

WARNING: Clang had build errors.                                                
- packages with build errors remain installed in c:\Users\phlavenk\JuliaData\v0
  .4  
- build the package(s) and all dependencies with `Pkg.build("Clang")`  
- build a single package by running its `deps/build.jl` script  

# 

'''

Seems to me LLVM libs are missing and are not downloaded as a Clang.jl dependecy.. Could you please tell me the steps to get LLVM running on windows so it's recognized by Julia Pkg ecosystem?
Thanks a lot.
Petr

Comment (bjarthur) [inline src/NIDAQ.jl:44]: you need to tell julia where the clang install is with

```
ENV["LLVM_CONFIG"]="/full/path/to/your/clang/installation/"
```

not sure where that would be on windows.  halfway down the Clang.jl README are very brief installation instructions.

**Comment by bjarthur** [comment]
@phlavenk i finally got Clang to work and have pushed the header files for NI-DAQmx v15.1.1 to NIDAQ.jl.  could you please modify your PR to use these for julia 0.4 and leave the v14 and v9.8 headers unmodified for use with julia v0.3?

**Comment by Petr-Hlavenka** [comment]
Thanks for the Clang work – I didn’t manage to compile the clang.jl beast on windows.
Do you think we should split the NIDAQ.jl and release new versions only for 0.4.x? In that case, no compat macro magic is necessary and maintenance is simpler. If this is the case I’ll prepare the 0.4-only PR.

**Comment by bjarthur** [comment]
yes, lets only support julia 0.4 from now on.  i've added a REQUIRE file to that effect already.  note that the Clang generated files rely on Compat, but i think we should modify NIDAQ.jl to not rely on it, as you have done already.

also, please submit the changes you made to test/ to support boards without certain functionality in a separate commit and PR.  that way there is more of a paper trail to document it.

thanks again for prodding the update to 0.4!  i'm sure many will find it useful.

**Comment by bjarthur** [comment]
support for julia 0.4 added in https://github.com/JaneliaSciComp/NIDAQ.jl/commit/5fd30b5489fe709a9b527cc13bc397ee4062954e

@phlavenk please submit the changes you made to test/ to support boards without certain functionality in a separate commit and PR

**Pull request closed without merging**